doc/fileexplorer.rst

```diff
@@ -1,23 +1,84 @@
+The **Files** pane is a filesystem and directory browser built right into Spyder.
+You can view and filter files according to their type and extension, open them with the :doc:`editor` or an external tool, and perform many common operations.
 
 .. image:: images/files/files-standard.png
-   :alt: Spyder File Explorer panel, showing a tree view of files and metadata
+   :alt: Spyder Files pane, showing a tree view of files and metadata
 
 
 
-==========================
-Performing file operations
-==========================
+===============
+File operations
+===============
 
-You can use context menus to access a number of functions on a per-file and per-directory basis, including running scripts, creating, renaming, moving, editing or deleting files, performing basic operations with ``git``, and opening an external file explorer.
+To browse the files on your system, you can use the arrows at the top of the pane.
+You can expand/collapse the folders in the pane to display the files and subdirectories hierarchically. 
+Double-clicking a folder will open it, showing the files inside and making it your working directory.
 
-.. image:: images/files/files-contextmenu-new.png
-   :align: center
-   :alt: File Explorer context menu, with common file manipulation tasks
+.. image:: images/files/files-browse.gif
+   :alt: Spyder Files pane showing browsing directories
+
+
+To open a file in the :guilabel:`Editor` from the :guilabel:`Files` pane, double-click its name.
+If you right-click over it, you will see a context menu that allows you to access a number of functions, including running scripts; creating, renaming, moving, deleting files; and opening them in your computer's file manager.
+
+.. image:: images/files/files-context-menu.png
+   :alt: Spyder Files pane showing context menu
+
+You can copy and paste one or several files to and from the pane, and copy their absolute or relative paths to the clipboard as text.
+If copying the paths for multiple files, they will be automatically formatted so you can paste them directly into a Python list.
+
+.. image:: images/files/files-copy-path.gif
+   :alt: Spyder Files pane showing copying absolute path to Editor
+
+Additionally, the :guilabel:`Files` pane allows you to perform basic operations with ``git``, like commiting your changes and browsing the repository a given file or folder is part of.
+
+.. image:: images/files/files-commit.gif
+   :alt: Spyder Files pane showing commiting change of file
+
+
+
+============
+Options menu
+============
+
+The options menu in the top right of the :guilabel:`Find` pane offers several ways to customize how your files are displayed.
+
+By default, the option :guilabel:`Show all files` is activated, which displays the contents of your working directory without filtering. 
+However, if you deactivate this, the pane will only show files matching the patterns set under :guilabel:`Show filenames with these extensions...`.
+
+.. image:: images/files/files-filters.gif
```

Comment:
I found it very hard to follow what happens here, because everything happened so quickly and the file structure was sufficiently complex that it was almost impossible to easily tell the effect of the actions taken. Initially enabling file filters happens almost instantly from the start of the GIF that its almost impossible to clearly see the before and after state, and thus the effect of the filters. Deleting the `.txt` extension and hitting `OK` had no immediate effect, as the two `txt` files were still there (though even this wasn't obvious, since they were hidden behind the dialog). File filters are then disabled and re-enabled to actually hide the files but again this all flashes by so quickly and there's so many other folders its hard to see what's going on without watching it several times. Finally, there appears to be a short black frame at the end

To fix this, I suggest:

* [ ] Use as the current directory a folder with perhaps a dozen files or so files different types, around half included in the default filter and half not, and perhaps half again of the former with the `.txt` extension, plus a couple subfolders. This will more clearly show the effect of the filters on a reasonable number of files.
* [ ] Slow down the actions some, and pause where needed: a second or two at the beginning before enabling filters, a second after closing the dialog, and another second at the end.
* [ ] If the issue of filters not getting applied immediately is not fixed, here in the text that the option needs to be toggled for changes to take effect.
* [ ] Remove black frame at the end

@ccordoba12 shouldn't the new filters be applied immediately when the user clicks OK? Its unintuitive for the user's changes to have no effect until manually disabling and re-enabling filters; if a fix can't be released promptly, i.e. connect the OK QButton to a Qt signal that refreshes the filter), we need to at least mention this in the docs.

Reply:
I agree with this suggestion too.

> @ccordoba12 shouldn't the new filters be applied immediately when the user clicks OK?

The filters are applied only when the filter button is clicked.

> we need to at least mention this in the docs.

Ok, sound good to me.

Reply:
> The filters are applied only when the filter button is clicked.

Yeah, what I explain above is that if filters are enabled (that button is toggled), changing the extensions in the dialog doesn't update the filter accordingly; the button has to be manually toggled on and off again to update them, which is quite unintuitive and may cause the user some frustration unless and until they figure out why there changes aren't being reflected in the filtered output. Therefore, the OK button should send a signal that's connected to whatever slot that triggers an update of the filters.

Reply:
Will redo with fewer files/folders and pane in the center to show only the pane area. Please ping me @ccordoba12 when the toggle and tooltip issues are fixed.

Reply:
Opened as spyder-ide/spyder#14328

Reply:
This is a lot better, thanks.